test/task.jl

```diff
@@ -65,6 +54,74 @@ end
     @test Ray.get_task_id() != task_id != subtask_id
 end
 
+@testset "object ownership" begin
+    @testset "unknown owner" begin
+        invalid_ref = ObjectRef(ray_jll.FromRandom(ray_jll.ObjectID))
+        @test !Ray.has_owner(invalid_ref)
+
+        msg = "An application is trying to access a Ray object whose owner is unknown"
+        @test_throws msg Ray.get_owner_address(invalid_ref)
+    end
+
+    @testset "driver get on worker put" begin
+        return_ref = submit_task(Ray.put, (2,))
+        remote_ref = Ray.get(return_ref)
+        @test Ray.has_owner(return_ref)
+        @test_broken Ray.has_owner(remote_ref)
+
+        # Convert address to string to compare
+        @test_broken begin
+            return_ref_addr = ray_jll.SerializeAsString(Ray.get_owner_address(return_ref))
+            remote_ref_addr = ray_jll.SerializeAsString(Ray.get_owner_address(remote_ref))
+            return_ref_addr != remote_ref_addr
+        end
+
+        @test_broken Ray.get(remote_ref) == 2
+    end
+
+    # TODO: When owner registration is enabled this test seems to corrupt other ObjectIDs
+    # such that later tests return the wrong results.
+    @testset "worker get on driver put" begin
+        local_ref = Ray.put(3)
+        return_ref = Ray.submit_task(Ray.get, (local_ref,))
+        @test return_ref != local_ref
+        @test Ray.has_owner(return_ref)
+        @test Ray.has_owner(local_ref)
+
+        # TODO: When owner registration is enabled this causes tasks to be re-run due to
+        # "lost objects"
+        @test_broken Ray.get(return_ref) == 3
```

Comment:
what does this produce currently?

Reply:
```
ERROR: on Ray task "Ray.get" with PID 89385: Attempted to wait for unowned object: ObjectRef("00ffffffffffffffffffffffffffffffffffffff0100000001000000")
```
The unowned object error comes from the worker